chaincode/contracts/AssetManager.sol

```diff
@@ -0,0 +1,235 @@
+        string reason
+    );
+    event TransferSuccesful(uint256 amount,
+        Enums.TokenType tokenType,
+        uint256 tokenId, 
+        address tokenAddress, 
+        address indexed from, 
+        address indexed to);
+    error TransferFailed(
+        uint256 amount,
+        Enums.TokenType tokenType,
+        uint256 tokenId, 
+        address tokenAddress, 
+        address from, 
+        address to,
+        string reason
+    );
+
+    address operator;
+    uint256 globalUserAuthLimit;
+    mapping(address => uint256) public userBalances;
+    // mapping(address => uint256) public userAuthorizations;
+    mapping(address => mapping(address => uint256)) private _allowances;
+
+    constructor(address operator_, uint256 globalUserAuthLimit_) {
+        setOperator(operator_);
+        globalUserAuthLimit = globalUserAuthLimit_;
+
+        }
+
+    modifier onlyOperator {
+        require(msg.sender == operator, "Can only be called by Operator");
+        _;
+    }
+
+    modifier onlyOwnerOrOperator {
+        bool ownerOperator = ((msg.sender == owner()) || (msg.sender == operator));
+        require(ownerOperator, "Can only be called by Owner or Operator");
+        _;
+    }
+
+    /**
+     * @dev Returns the address of the current operator.
+     */
+
+    function setOperator(address operator_) onlyOwnerOrOperator public {
+        operator = operator_;
+    }
+
+    function deposit() public payable {
+        userBalances[address(msg.sender)] += msg.value;
+        // update the userBalance
+        emit DepositSuccesful(
+            msg.sender,
+            msg.value,
+            userBalances[address(msg.sender)]
+        );
+    }
+
+        function withdraw(uint256 amount) public payable {
+        uint256 balance =  userBalances[address(msg.sender)];
+        // if zero is passed withdraw all funds
+        if (amount == 0){ amount = balance; }
+        // check msg.senders balance
+        if (amount > balance) {
+            revert WithdrawalFailed(
+                msg.sender,
+                amount,
+                balance,
+                "Insufficient Locked Funds to Withdraw "
+            );
+        }
+
+        // withdraw funds from the contract (update userBalance before transfer to protect from reentracy attack)
+        uint256 newBalance = balance - amount;
+        userBalances[address(msg.sender)] = newBalance;
+        payable(msg.sender).transfer(amount);
+
+        // update the userBalance
+        emit WithdrawalSuccesful(
+            msg.sender,
+            amount,
+            userBalances[address(msg.sender)]
+        );
+    }
+
+    function allowance(address owner, address spender) public view returns (uint256) {
+        return _allowances[owner][spender];
+    }
+
+    function approve(address spender, uint256 amount) public returns (bool) {
+        address owner = _msgSender();
+        _approve(owner, spender, amount);
+        return true;
+    }
+    function _approve(
```

Comment:
We don't need two functions here. Token contracts had it to allow overrides

Reply:
done